### PR TITLE
[JR-1279] Hide a company name on a job list and details pages for a public and restricted anonymous

### PR DESCRIPTION
**Pull request by wwebec**

### CODE REVIEW

File: src/main/webapp/app/job-search/job-detail/job-detail.component.html [hunk 1]
```diff
@@ -5,44 +5,51 @@
...
                     <hr class="visible-print-block d-none"/>
 
-                    <div class="alv-logo-box logo-box logo-box__company flex-height md-whiteframe-1dp" >
-                        <div class="detail__content" *ngIf="job.jobContent.company as company">
-                            <address>
-                                <strong>{{ company.name }}</strong>
-                                <br>
+                    <div
+                        class="alv-logo-box logo-box logo-box__company flex-height md-whiteframe-1dp"
+                        *ngIf="!job.publication.publicAnonymous && !job.publication.restrictedAnonymous;then showLeftCompanyBox">
```
Comment (cjpsycos) [inline src/main/webapp/app/job-search/job-detail/job-detail.component.html]: ngIf is enough. if there is no else, using template is overdone.
move the logic in the component as flag companyAnonymous

File: src/main/webapp/app/job-search/job-detail/job-detail.component.html [hunk 2]
```diff
@@ -56,73 +63,103 @@
...
                        class="d-print-none">
                 {{'job-detail.external-job-disclaimer' | translate}}
             </ngb-alert>
 
             <div class="detail__title">
-                <h2 class="text--light" [innerHTML]="jobDescription.title | safeHtml"></h2>
-                <span class="text--primary-color">{{ job.jobContent.company.name }}</span>
+                <h2 class="text--light"
+                    [innerHTML]="jobDescription.title | safeHtml"></h2>
+                <span class="text--primary-color" *ngIf="job.publication.publicAnonymous || job.publication.restrictedAnonymous;then hideCompanyName else showCompanyName"></span>
```
Comment (cjpsycos) [inline src/main/webapp/app/job-search/job-detail/job-detail.component.html]: use companyAnonymous flag

File: src/main/webapp/app/job-search/job-search-list-item/job-search-list-item.component.html [hunk 3]
```diff
@@ -7,8 +7,11 @@ <h4 class="text--primary-color order-2 order-sm-1"><span [innerHtml]="jobDescrip
     </div>
 
     <div>
-        <span class="text--primary-color">{{job.jobContent.company.name}}</span>
-        <jr2-job-locality [location]="job.jobContent.location"></jr2-job-locality>
+        <span class="text--primary-color" *ngIf="job.publication.publicAnonymous || job.publication.restrictedAnonymous;then hideCompanyName else showCompanyName"></span>
```
Comment (cjpsycos) [inline src/main/webapp/app/job-search/job-search-list-item/job-search-list-item.component.html]: move the logic in the component as flag companyAnonymous

Comment (wwebec) [reply src/main/webapp/app/job-search/job-search-list-item/job-search-list-item.component.html]: moved

File: src/main/webapp/app/job-search/job-detail/job-detail.component.html (hunk 1 above)
Comment (wwebec) [reply src/main/webapp/app/job-search/job-detail/job-detail.component.html]: moved

File: src/main/webapp/app/job-search/job-detail/job-detail.component.html (hunk 2 above)
Comment (wwebec) [reply src/main/webapp/app/job-search/job-detail/job-detail.component.html]: done

**Pull request merged**